plugins/project-init/scripts/project_state.sh

```
@@ -194,6 +194,83 @@ GI_SERENA=$(ignored .serena/)
+# 나머지는 버린다 (필드 병합 없음). 두 정의가 갈라지면 한쪽 편집이 조용히 죽는다.
+# 고아 등록(삭제된 플러그인의 서버)은 사용 이력이 있어야 알 수 있어 여기서 다루지 않는다.
+# 손상된 JSON 에서 jq 는 non-zero 로 끝난다. `set -euo pipefail` 아래서 그 실패는
+# 명령 치환을 타고 올라와 스크립트 전체를 죽인다 — 중복 하나 못 세는 대신 진단이
+# 통째로 사라진다. 파싱 실패를 이 축 안에 가두고, 대신 unreadable 로 실어 보낸다:
+# 조용히 "중복 없음" 이라고 답하는 것이 제일 나쁘다.
+mcp_keys() { jq -r '.mcpServers // {} | keys[]' "$1" 2>/dev/null | sort || true; }
+mcp_broken() { [ -f "$1" ] && [ -s "$1" ] && ! jq empty "$1" >/dev/null 2>&1; }
+MCP_USER=0; MCP_SETTINGS=0; MCP_DUPES='[]'
+_bad=""
+for f in "$HOME/.claude.json" "$HOME/.claude/settings.json"; do
+  if mcp_broken "$f"; then _bad="${_bad}${f#$HOME/}"$'\n'; fi
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)